File: HourRank_31_P2.cpp
// code_report Solution
// 

void solve(int n, vector<int> a) {
    sort(begin(a), end(a), greater<int>());
    auto on  = accumulate(begin(a), begin(a) + n, 0.0);
    auto off = accumulate(begin(a) + n, end(a), 0.0);
    auto i   = 0;
    auto avg = [&]() { return (off + on) / (n - i); };
    while (avg() < a[i] && i < n) on -= a[i], ++i;
    cout << setprecision(10) << fixed << avg() << endl;
}

File: HourRank_31_P2.py
# code_report Solution
# https://youtu.be/E6_-mvn8xIw

def avg(a, b):
    return (sum(a)+b)/len(a)

def solve(n, a):
    a = sorted(a, reverse = True)
    on, off = a[0:n], sum(a[n:])
    while on and avg(on, off) < max(on): on.pop(0)
    print("%.5f"% (avg(on, off)))
